rust/headless-client/src/imp_windows.rs

```diff
@@ -39,18 +54,119 @@ pub(crate) fn default_token_path() -> std::path::PathBuf {
     PathBuf::from("token.txt")
 }
 
+/// Only called from the GUI Client's build of the IPC service
+///
+/// On Windows, this is wrapped specially so that Windows' service controller
+/// can launch it.
 pub fn run_only_ipc_service() -> Result<()> {
+    windows_service::service_dispatcher::start(SERVICE_NAME, ffi_service_run)?;
+    Ok(())
+}
+
+// Generates `ffi_service_run` from `service_run`
+windows_service::define_windows_service!(ffi_service_run, windows_service_run);
+
+fn windows_service_run(_arguments: Vec<OsString>) {
+    if let Err(_e) = fallible_windows_service_run() {
+        todo!();
+    }
+}
+
+#[cfg(debug_assertions)]
+const SERVICE_RUST_LOG: &str = "debug";
+
+#[cfg(not(debug_assertions))]
+const SERVICE_RUST_LOG: &str = "info";
+
+// Most of the Windows-specific service stuff should go here
+fn fallible_windows_service_run() -> Result<()> {
     let cli = Cli::parse();
-    let (layer, _handle) = cli.log_dir.as_deref().map(file_logger::layer).unzip();
-    setup_global_subscriber(layer);
+    let log_path =
+        crate::known_dirs::ipc_service_logs().context("Can't compute IPC service logs dir")?;
+    std::fs::create_dir_all(&log_path)?;
+    let (layer, _handle) = file_logger::layer(&log_path);
+    let filter = EnvFilter::from_str(SERVICE_RUST_LOG)?;
```

Comment:
We could. This function only runs when Windows is running the IPC service, so it's not interactive, the variable would ultimately come from a file or the registry.

I checked these sources
- https://stackoverflow.com/questions/305870/accessing-environment-variables-from-windows-services
- https://learn.microsoft.com/en-us/windows/win32/services/service-startup

> The service control manager does not support passing custom environment variables to a service at startup. Also, the service control manager does not detect and pass on changes to environment variables as the service is running. Instead of making a service dependent on an environment variable, use registry values or [ServiceMain](https://learn.microsoft.com/en-us/windows/win32/api/winsvc/nc-winsvc-lpservice_main_functiona) arguments.

It's weird.